### ISSUE TITLE
Don't require consent if consent already given

### ISSUE THREAD

**Issue by tdg** [swapmyvote/swapmyvote#245]
The consent box needs ticking each time a returning user logs in

**Comment by aspiers** [comment]
This requires a cookie if client-side.

**Comment by egglestn** [comment]
Consent does create a cookie, but it doesn't show up (including in debugger) until the next turbolinks refresh, so there is JS code that hides the cookie consent message, it only lasts until the next page refresh. At that point, the cookie-consent cookie should take over. 

If you're not seeing the cookie being saved, please check you don't have anything blocking cookies.

The cookie does rely on a matching domain, testing just now I have a _swapmyvote_cookie_consent cookie set successfully in prod, https://www.swapmyvote.uk/ It's possible that the cookie or the JS is not working for all the QA domains, or all browsers, if so let me know what you were using when you saw the issue

**Comment by aspiers** [comment]
This is not about consent for cookies, it's about the privacy consent
checkbox in the login modal. We need a new cookie for that.

**Comment by egglestn** [comment]
Oops, sorry. Will pick it up when I can

**Comment by stevebaxter** [comment]
We might want to consider whether we should do this. We don't know *for certain* that this is the same person, for instance what if it's a shared machine (public or between people in a household). For GDPR it's important that consent is unambiguous, that means we must never pre-tick a checkbox like this.

@tdg @aspiers ?

**Comment by tdg** [comment]
You're right

Let's de-prioritise this to after the election

**Comment by aspiers** [comment]
Ugh yeah, so it has to be server-side :-(  At least noone moaned about it yet.

**Comment by baob** [comment]
Looks like this overlaps with https://github.com/swapmyvote/swapmyvote/issues/345 . Do they duplicate each other ?

**Comment by aspiers** [comment]
No, not at all.  That's about consent to share email with swap partner; this is about consent to process PII.  However it *does* overlap with #591.
